scripts/utilities/update_revision.sh

```diff
@@ -139,61 +108,58 @@ port ${GERRIT_PORT} with user \"${GERRIT_USER}\"."
-		if [[ "${topic_status[1]}" -eq -1 ]]; then
-			print_msg 0 "Change with review label -1. \
-Aborting the update.";
-			return 0
-		fi
-	else
-		print_msg 0 "Not found any pending change in topic: ${topic}."
-	fi
+
+	local target_remote=git@github.com:${GITHUB_USER}/advance-toolchain.git
+	local target_branch=auto-update_${cfg}_${pkg}
 
 	# Save current branch and switch to a work branch
 	#
-        current_branch=$(git rev-parse --abbrev-ref HEAD)
-	work_branch=${topic}
-        # -B option resets the branch if it already exist.
-        git checkout -B ${work_branch}
-
-	# Check gerrit's hook exists so that a change-id is generated
-	# Otherwise download the hook script
-	gitdir=$(git rev-parse --git-dir)
-	if [[ ! -f ${gitdir}/hooks/commit-msg ]]; then
-		scp -p -P ${GERRIT_PORT} ${GERRIT_USER}@${GERRIT_SERVER}:\
-hooks/commit-msg ${gitdir}/hooks/
+	current_branch=$(git rev-parse --abbrev-ref HEAD)
+	# -B option resets the branch if it already exists.
+	git checkout -B ${target_branch}
+
+	# Here we check if the new sources file we generated by updating the
+	# revision is different from the sources file at the last commit
+	# on the appropriate branch.
+	# To do that, we run a git diff between our working tree and the
+	# appropriate branch. Since the branch we want to compare to is
+	# remote, we need to add a new temporary remote to the repo.
+	git remote add TMP_DIFF ${target_remote}
+	git fetch TMP_DIFF
+
+	print_msg 2 "Checking ${pkg} revision on the topic branch"
+
+	if [[ $(git ls-remote -h ${target_remote} | grep -c ${target_branch}) = 1 ]] \
+		 && git diff --exit-code TMP_DIFF/${target_branch} -- ${1};
+	then
+		print_msg 0 "$pkg revision is already up to date, new commit not necessary"
+
+		git remote remove TMP_DIFF
+
+		git checkout ${current_branch} ${1}
+		git checkout ${current_branch}
+		return 0
+	else
+		print_msg 0 "$pkg revision is outdated, continuing commit..."
 	fi
 
-        print_msg 2 "Generating a patch";
-        file=$(basename $(dirname ${1}))/$(basename ${1})
+	git remote remove TMP_DIFF
+
+	print_msg 2 "Generating a patch";
+	file=$(basename $(dirname ${1}))/$(basename ${1})
 
-        git add ${1}
-        local msg="Update ${pkg} on AT ${cfg}\n\
+	git add ${1}
+	local msg="Update ${pkg} on AT ${cfg}\n\
 Bump to revision ${2}\n\n"
-	if [[ ! -z "${change_id}" ]]; then
-		msg+="Change-Id: ${change_id}"
-		print_msg 0 "Reuse Change-Id of pending change on Gerrit"
-	else
-		print_msg 0 "Generate commit with a new Change-Id"
-	fi
-        echo -e ${msg} | git commit -F -
+	echo -e ${msg} | git commit -F -
 
-	# Finally send to gerrit
+	# Finally send to GitHub
 	# Use topic branch to keep track of changes
-	print_msg 2 "Sending commit to Gerrit"
-        git push ssh://${GERRIT_USER}@${GERRIT_SERVER}:${GERRIT_PORT}\
-/advance-toolchain HEAD:refs/for/master%topic=${topic}
+	print_msg 2 "Sending commit to GitHub"
+	git push --force ${target_remote} HEAD:${target_branch}
```

Comment:
In Gerrit, when you push a patch to for/master, that means you're creating pull request.
On Github, the steps are different: after pushing the patch, it's necessary to open a pull request too.